### ISSUE TITLE
Reinstating <biblStruct> in consolidated schema

### ISSUE THREAD

**Issue by eifionjones** [msDesc/consolidated-tei-schema#35]
Would it be possible to reinstate <biblStruct> into the consolidated schema? A number of our projects store their bibliographies in Zotero and export them into TEI to embed in a `<listBibl>` - and these come out as `<biblStruct>`

**Comment by andrew-morrison** [comment]
I notice that @yf227 has added `biblStruct` to a new Fihrist record:

https://github.com/fihristorg/fihrist-mss/blob/03d98b57446259ffa7ee0db04f88d5b296eaeb96/collections/cambridge%20university/Or_1878.xml#L142-L162

There are no XSLT templates for displaying this yet, so it just gets dumped as a string of unformatted text (except titles are italicized) in the Bibliography:

https://www.fihrist.org.uk/catalog/manuscript_16739

@holfordm has added biblStruct and its child elements to the ODD. But compiling that to Relax NG and testing reveals there are few changes which are needed for it to be valid:

 * If the bibliographic reference is to a book, the `biblStruct` must contain a `monogr` child element
 * Publication information within that must be grouped into an `imprint` element.

So the example above becomes:
```
<listBibl>
    <biblStruct type="book" xml:id="Ullendorff1961" corresp="http://zotero.org/users/local/VPTDBJud/items/2GNGMD9M">
        <monogr>
            <title level="m">Catalogue of Ethiopian manuscripts in the Cambridge University Library
                / by Edward Ullendorff and Stephen G. Wright ; with a contribution by D.A.
                Hubbard.</title>
            <author>
                <forename>Edward</forename>
                <surname>Ullendorff</surname>
            </author>
            <imprint>
                <pubPlace>Cambridge</pubPlace>
                <publisher>Cambridge University Press</publisher>
                <date>1961</date>
            </imprint>
        </monogr>
    </biblStruct>
    <biblStruct type="book" xml:id="Leslau1951" corresp="http://zotero.org/users/local/VPTDBJud/items/ATRQKIFH">
        <monogr>
            <title level="m">Falasha anthology / translated from Ethiopic sources with an introd. by
                Wolf Leslau.</title>
            <author>
                <forename>Wolf</forename>
                <surname>Leslau</surname>
            </author>
            <imprint>
                <pubPlace>New Haven</pubPlace>
                <publisher>New Haven : Yale University Press, 1951.</publisher>
                <date>1951</date>
            </imprint>
        </monogr>
    </biblStruct>
</listBibl>
```
Will this work for your systems?

**Comment by yf227** [comment]
Hi Andrew

Thanks for this.
It would be perfect to have the bibliography in valid TEI as Zotero exports it. I fiddled with it too much trying to get it to validate against the schema that it became corrupted.
Huw says they have done the XSLT coding for <BiblStruct> and he will share it with you.
Shall I correct the record and submit it again?

ysmn

Yasmin Faghihi
Head of Near and Middle Eastern Department
Manuscripts and Printed Collections
Cambridge University Library
West Road
CB3 9DR UK
Tel.: +44 1223 764739
Email: yf227@cam.ac.uk<mailto:yf227@cam.ac.uk>
Chair
Middle East Libraries COMmittee (MELCOM) UK
http://www.melcom.org.uk/
FIHRIST
http://www.fihrist.org.uk/home
--

**Comment by andrew-morrison** [comment]
Yes, you will be able to just pick the export as TEI option in Zotero and paste it in, once I've done a new release on the msdesc schema later today.

I'll let you know when I have done that, then you can fix Or. 1878 (and Add. 1570 which I see you added yesterday) and will be able to validate in Oxygen.

**Comment by yf227** [comment]
Excellent! Thanks Andrew!
I am still working on Add. 1570…it’s a complex record…
Looking forward to seeing the update.
ysmn

Yasmin Faghihi
Head of Near and Middle Eastern Department
Manuscripts and Printed Collections
Cambridge University Library
West Road
CB3 9DR UK
Tel.: +44 1223 764739
Email: yf227@cam.ac.uk<mailto:yf227@cam.ac.uk>
Chair
Middle East Libraries COMmittee (MELCOM) UK
http://www.melcom.org.uk/
FIHRIST
http://www.fihrist.org.uk/home
--

**Comment by andrew-morrison** [comment]
There were some issues doing the new release of the schema, but it is done now. You should be able to copy and paste Zotero's TEI export straight into the `listBibl` in `additional`.

You may have to restart Oxygen to force it to download a fresh copy of the schema file.

**Comment by yf227** [comment]
Waw, excellent!
Thank you, Andrew!!
Will check it out straight away.



Yasmin Faghihi
Head of Near and Middle Eastern Department
Manuscripts and Printed Collections
Cambridge University Library
West Road
CB3 9DR UK
Tel.: +44 1223 764739
Email: yf227@cam.ac.uk<mailto:yf227@cam.ac.uk>
Chair
Middle East Libraries COMmittee (MELCOM) UK
http://www.melcom.org.uk/
FIHRIST
http://www.fihrist.org.uk/home
--

**Comment by eifionjones** [comment]
Thanks Andrew, brilliant!
Best wishes,
Huw

**Comment by yf227** [comment]
It works! ☺

Yasmin Faghihi
Head of Near and Middle Eastern Department
Manuscripts and Printed Collections
Cambridge University Library
West Road
CB3 9DR UK
Tel.: +44 1223 764739
Email: yf227@cam.ac.uk<mailto:yf227@cam.ac.uk>
Chair
Middle East Libraries COMmittee (MELCOM) UK
http://www.melcom.org.uk/
FIHRIST
http://www.fihrist.org.uk/home
--

**Comment by andrew-morrison** [comment]
Reopening this issue because biblStruct needs new XSLT to display it in HTML. Yasmin mentioned above there already being some, which you can send me, Huw. Or do you want to do it as a merge request?

**Comment by eifionjones** [comment]
Hi Andrew,

I’m attaching the code we use to turn biblStruct into html

For context – this is not an independent piece of XSLT – it’s a prefilter embedded in XTF and contains the whole chunk of code we use to turn TEI into what gets consumed by our viewer – so you can ignore most of it.

We do however turn the <biblStruct> into html – so that bit might prove useful to you

It starts on line 4692 (commented “BIBLIOGRAPHY PROCESSING”) and continues to the end

The first three templates just look for bibliographies and put a wrapper round them

The rest goes through a lot of logic to turn the contents of <biblStruct> into html

This was all written by Pete Johnston, a former colleague, so I didn’t go through the whole process myself, but if there are questions do ask!

Hope all well with you.

Best,

Huw
